File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.cpp
class Solution {
public:
    int findMin(vector<int>& nums) {
        int n = nums.size();
        int low = 0;
        int high = n-1;
        int mini = INT_MAX;
        
        while(low <= high){
            int mid = (low + high)/2;
            // left sorted
            if(nums[low]<= nums[mid]){
                mini = min(mini,nums[low]);
                low = mid+1;
            }
            // right sorted
            else{
                mini = min(mini,nums[mid]);
                high = mid-1;
            }
        }
        return mini;
    }
};

File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        n = len(nums)
        nums.sort()
        return nums[0]
        # left = 0
        # right = n-1


        # while left <= right:
        #     mid = (left+ right) // 2
        #     if(nums[mid] )